### ISSUE TITLE
Cannot compile OTP in Windows 11 with WSL 2 & VS2019

### ISSUE THREAD

**Issue by trippleflux** [erlang/otp#5291]
**Describe the bug**
I have tried to compile maint-24 in Windows 11 with WSL 2,UBUNTU 20.04 LTS & VS2019 16.11.5, the first issue that I have found
the otp sources folder cannot in long path, Was tried on "C:\Users\username\Works\Compiling\OTP" but failed.
Moving the OTP folder into such as "C:\opt\OTP" is get me further but later on will failed on "otp_build boot -a" with error code 255 , been searching around but seems to be a generic error code.

**To Reproduce**
1. Clone the git "git clone https://github.com/erlang/otp.git -b maint-24" into "C:\opt"
2. add bash ".profile" with following content (specifically made for VS 2019 16.11 x64 Enterprise)
```
#Erlang Build
C_DRV=/mnt/c
# Program Files
PRG_FLS64=$C_DRV/Program\ Files
PRG_FLS32=$C_DRV/Program\ Files\ \(x86\)

# Visual Studio
VISUAL_STUDIO_ROOT=$PRG_FLS32/Microsoft\ Visual\ Studio/2019
WIN_VISUAL_STUDIO_ROOT="C:\\Program Files (x86)\\Microsoft Visual Studio\\2019"

# SDK
SDK=$PRG_FLS32/Windows\ Kits/10
WIN_SDK="C:\\Program Files (x86)\\Windows Kits\\10"
MICROSOFT_SDK="C:\\Program Files (x86)\\Microsoft SDKs\\Windows Kits\\10"
NET_FRAMEWORK="C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319"
NETFX_SDK="C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.8"
MONO=$PRG_FLS32/Mono/bin

# NSIS
NSIS_BIN=/mnt/c/Users/username/Documents/Tools/NSIS
NASM_BIN=/mnt/c/Users/username/Documents/Tools/Nasm/x64

# Java 
FOP_BIN=/mnt/c/Users/username/Documents/Tools/fop-2.1
JAVA_BIN=$PRG_FLS64/Java/jdk-17/bin
CMAKE_BIN=/mnt/c/Users/username/Documents/Tools/cmake-3.21.2-windows-i386/bin

## The PATH variable should be Cygwin'ish
VCPATH=\
$VISUAL_STUDIO_ROOT/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:\
$VISUAL_STUDIO_ROOT/Enterprise/Common7/IDE/VC/VCPackages:\
$VISUAL_STUDIO_ROOT/Common7/IDE:\
$VISUAL_STUDIO_ROOT/Common7/Tools:\
$SDK/bin/x64

## Microsoft SDK libs
LIBPATH=$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\ATLMFC\\lib\\x64:\
$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\lib\\x64:\
$WIN_SDK\\UnionMetadata\\10.0.19041.0:\
$WIN_SDK\\References\\10.0.19041.0:\
$WIN_SDK\\Lib\\10.0.19041.0\\um\\x64:\
$WIN_SDK\\Lib\\10.0.19041.0\\ucrt\\x64:\
$NET_FRAMEWORK

LIB="$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\ATLMFC\\lib\\x64;\
$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\lib\\x64;\
$WIN_SDK\\UnionMetadata\\10.0.19041.0;\
$WIN_SDK\\References\\10.0.19041.0;\
$WIN_SDK\\Lib\\10.0.19041.0\\um\\x64;\
$WIN_SDK\\Lib\\10.0.19041.0\\ucrt\\x64;\
$NET_FRAMEWORK"


INCLUDE="$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\ATLMFC\\include;\
$WIN_VISUAL_STUDIO_ROOT\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\include;\
$NETFX_SDK\\include\\um;\
$WIN_SDK\\include\\10.0.19041.0\\ucrt;\
$WIN_SDK\\include\\10.0.19041.0\\shared;\
$WIN_SDK\\include\\10.0.19041.0\\um;\
$WIN_SDK\\include\\10.0.19041.0\\winrt"

# Put nsis, c compiler and java in path
export PATH=$VCPATH:$PATH:$JAVA_BIN:$NSIS_BIN:$FOP_BIN:$CMAKE_BIN:$NASM_BIN:$MONO

# Make sure LIB and INCLUDE is available for others
export LIBPATH LIB INCLUDE
```
3. Compile OpenSSL 1.1.1 and WXWidgets  (Follow https://github.com/erlang/otp/blob/maint-24/HOWTO/INSTALL-WIN32.md )
4. From Ubuntu prompt, type in : 
```
ERL_TOP=/mnt/c/opt/otp
export ERL_TOP
cd $ERL_TOP
eval `./otp_build env_win32 x64`
./otp_build configure --with-javac --with-ssl --enable-threads
```
5. and then type in "./otp_build boot -a", after early process of this, it will error out with 255 error code.

**Expected behavior**
OTP compilation should be possible in any Windows version

**Affected versions**
Tested : 
```
Windows 10 21H2 (19044.1288) with OTP maint-24 compilation is working.
Windows 11 21H2 (22000.228) with OTP maint-24 compilation is not working.
```
So compilation for now only working in Windows 10 variant.

**Comment by JeromeDeBretagne** [comment]
> As stated in the readme we only use on WSL-1,  I have not been able to make WSL-2 work and reported the bug to microsoft,
> 
> Is it not possible to install WSL-1 on windows-11, or is just the default WSL-2?

WSL 1 is still available on Windows 11, this is mentioned in the FAQ here for instance :

https://devblogs.microsoft.com/commandline/a-preview-of-wsl-in-the-microsoft-store-is-now-available/

For how long, that's another question...

**Comment by fire** [comment]
To be clear Windows 11 cannot compile erlang or was there a workaround?

**Comment by JeromeDeBretagne** [comment]
> To be clear Windows 11 cannot compile erlang or was there a workaround?

According to [comment #4](https://github.com/erlang/otp/issues/5291#issuecomment-949209458) from @trippleflux , **compiling Erlang works on Windows 11 as long as you use WSL 1** (Windows Subsystem for Linux, version 1) using the following command : `wsl --set-default-version 1`


Making it work also when using WSL 2 (instead of WSL 1) is the topic of this issue. It is not supported yet, that would be a nice enhancement.

**Comment by trippleflux** [comment]
@fire 
Just compiled maint-24 or current 24.2.2 with WSL-1 on Windows 11 x64 22000.527 with VS 2022, working fine for me.

For use with vs 2022 compilation, a quote from my notes : 
##Edit .\otp\erts\etc\win32\wsl_tools\SetupWSLcross.bat
```
IF EXIST "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat". (
   call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" %~1 > nul
   goto continue
)
```

**Comment by dgud** [comment]
Closing this since as far as I know there is only wsl2 (microsoft) issues that blocks this, not anything in our build steps.
Also accessing windows disk from wsl 2 is horrible slow, I just tested doing a git status on an Erlang/OTP repo on the windows disk and it takes several minutes.  And as far as I know we need to build on the windows disk the windows compiler doesn't work with files located on the wsl file system.
